### DISCUSSION TITLE
I need a client in Node.js or at least the protocol for implementing such a client

### DISCUSSION THREAD

**Discussion by ariel42** [erebe/wstunnel#375] [Q&A]
Hello,
Thank you for this great project!

Are there clients in other language, specifically in Node.js? I want to integrate the reverse tunneling client into my Node.js application without relying on other running process in the machine, and I can't find a sample implementation or a document describing the protocol flow so I can implement this by myself.

Thank you.

**Comment by erebe** [comment]
Hello,

everyting is there for websocket https://github.com/erebe/wstunnel/blob/main/src/tunnel/transport/websocket.rs#L246

The tunnel information is passed in SEC_WEBSOCKET_PROTOCOL header, inside a jwt.

For reverse tunnel, if your tunnel is dynamic, ti the response where to connect to is going to be forwarded back inside the COOKIE header https://github.com/erebe/wstunnel/blob/main/src/tunnel/client/client.rs#L177
